hcl2template/types.packer_config.go

```diff
@@ -320,41 +320,53 @@ func (cfg *PackerConfig) getCoreBuildProvisioners(source SourceUseBlock, blocks
 	var diags hcl.Diagnostics
 	res := []packer.CoreBuildProvisioner{}
 	for _, pb := range blocks {
-		if pb.OnlyExcept.Skip(source.String()) {
-			continue
-		}
-		provisioner, moreDiags := cfg.startProvisioner(source, pb, ectx)
+		skip, coreBuildProv, moreDiags := cfg.getCoreBuildProvisioner(source, pb, ectx)
 		diags = append(diags, moreDiags...)
-		if moreDiags.HasErrors() {
+		if moreDiags.HasErrors() || skip {
 			continue
 		}
+		res = append(res, coreBuildProv)
+	}
+	return res, diags
+}
 
-		// If we're pausing, we wrap the provisioner in a special pauser.
-		if pb.PauseBefore != 0 {
-			provisioner = &packer.PausedProvisioner{
-				PauseBefore: pb.PauseBefore,
-				Provisioner: provisioner,
-			}
-		} else if pb.Timeout != 0 {
-			provisioner = &packer.TimeoutProvisioner{
-				Timeout:     pb.Timeout,
-				Provisioner: provisioner,
-			}
+func (cfg *PackerConfig) getCoreBuildProvisioner(source SourceUseBlock, pb *ProvisionerBlock, ectx *hcl.EvalContext) (skip bool, coreBuildProv packer.CoreBuildProvisioner, diags hcl.Diagnostics) {
```

Comment:
Nit: I had to read the function a few times before I figured out that you had moved to named returns. I can see now that with the skip variable why you might have done it. But is was confusing to me at first. I think a comment for this method would be better suited if the skip return is needed. 

We have some guidance in our style guide about this https://eng-handbook.hashicorp.services/language-style/go/#return-values

Reply:
I'm moving the skip outside this method so that won't be a problem anymore